packages/core-mobile/app/screens/bridge/hooks/useTokenForBridgeTransaction.ts

```diff
@@ -9,43 +8,54 @@ import { NetworkContractToken } from '@avalabs/vm-module-types'
 import { useNetworkContractTokens } from 'hooks/networks/useNetworkContractTokens'
 import { useNetworks } from 'hooks/networks/useNetworks'
 import { useMemo } from 'react'
+import { isBitcoinChainId } from 'utils/network/isBitcoinNetwork'
+import { getChainIdFromCaip2 } from 'temp/caip2ChainIds'
+import { isEthereumChainId } from 'services/network/utils/isEthereumNetwork'
 
 export function useTokenForBridgeTransaction(
-  bridgeTransaction: BridgeTransaction | BridgeTransfer | undefined,
+  bridgeTransaction: BridgeTransfer | undefined,
   isTestnet: boolean
 ): NetworkContractToken | NetworkToken | undefined {
   const chainId = useMemo(() => {
-    switch (bridgeTransaction?.sourceChain) {
-      case Blockchain.BITCOIN:
-        return isTestnet ? ChainId.BITCOIN_TESTNET : ChainId.BITCOIN
-      case Blockchain.ETHEREUM:
-        // ETHEREUM_SEPOLIA doesn't have contract tokens, so always use ETHEREUM_HOMESTEAD chainid.
-        return ChainId.ETHEREUM_HOMESTEAD
-      case Blockchain.AVALANCHE:
-      default:
-        return isTestnet
-          ? ChainId.AVALANCHE_TESTNET_ID
-          : ChainId.AVALANCHE_MAINNET_ID
+    const sourceChainId = bridgeTransaction?.sourceChain.chainId
+      ? getChainIdFromCaip2(bridgeTransaction.sourceChain.chainId)
+      : undefined
+
+    if (!sourceChainId) return undefined
+
+    if (isBitcoinChainId(sourceChainId)) {
+      return isTestnet ? ChainId.BITCOIN_TESTNET : ChainId.BITCOIN
+    }
+
+    if (isEthereumChainId(sourceChainId)) {
+      return ChainId.ETHEREUM_HOMESTEAD
     }
+
+    return isTestnet
+      ? ChainId.AVALANCHE_TESTNET_ID
+      : ChainId.AVALANCHE_MAINNET_ID
   }, [bridgeTransaction, isTestnet])
 
   const { getNetwork } = useNetworks()
   const network = getNetwork(chainId)
-  if (!network) {
-    throw new Error(`Network not found for chainId: ${chainId}`)
-  }
-
   const tokens = useNetworkContractTokens(network)
+  const symbol = bridgeTransaction?.asset.symbol
 
   return useMemo(() => {
-    const token = tokens.find(t => t.symbol === bridgeTransaction?.symbol)
+    const token = tokens.find(t => t.symbol === symbol)
 
     if (token) return token
 
-    if (bridgeTransaction?.symbol === BITCOIN_NETWORK.networkToken.symbol) {
+    if (symbol === BITCOIN_NETWORK.networkToken.symbol) {
       return BITCOIN_NETWORK.networkToken
+    } else if (
+      chainId &&
+      isEthereumChainId(chainId) &&
+      network?.networkToken.symbol === symbol
+    ) {
+      return network?.networkToken
     }
```

Comment:
just curious why do we need this logic? won't it always just be ethereum token?

Reply:
removed